app/models/form526_submission.rb

```diff
@@ -126,9 +126,14 @@ def submit_with_birls_id_that_hasnt_been_tried_yet!(
     log_exception_to_sentry e, extra_content_for_sentry
   end
 
+  # Note that the User record is stored in Redis temporarily, for 30 minutes: User.redis_namespace_ttl => 1800
+  # Checks against the User record first, and then resorts to checking the auth_headers
+  # for the name attribute if the User record doesn't exist or doesn't contain the first name
   def get_first_name
     user = User.find(user_uuid)
-    user&.first_name&.upcase
+    return user&.first_name&.upcase if user&.first_name.present?
+
+    auth_headers&.dig('va_eauth_firstName')&.upcase
```

Comment:
Mimicking the `full_name` method, use the `auth_headers` in case `user` or `user.first_name` is nil

Reply:
Why not just use auth_headers since it's more reliable? Is it because it's not the canonical source?

Reply:
Not sure, but I don't want to make a change that conflicts with current behaviors that may expect info from the `user` record, which is created upon [user session creation](https://dsva.slack.com/archives/C01BVF2A3V0/p1645818462607279?thread_ts=1645807137.696329&cid=C01BVF2A3V0). Just playing it safe.

Reply:
I have no idea why we don't use the name from the auth_headers all the time......I do know we have issues with names when submitting claims.

Originally, get_first_name was called within the scope of the web user's submit 526 call, at this point the code is synchronous, the user has to be logged in to make the submit call, so the calling user has a current session, and User.find should work.

However, after the fasttrack changes, get_first_name is called in a method in a place that is outside of the users initial 526 submit call.  Most of the time, it's well within the redis 30 minute eject time, but it is not guaranteed since now the call is behind the asynchronous fasttrack sidekiq job.

All that being said, this new version of get_first_name is a good idea since it is now possible that this method get called after the user session has expired.